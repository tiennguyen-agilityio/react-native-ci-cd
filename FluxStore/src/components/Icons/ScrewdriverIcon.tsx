import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const ScrewdriverIcon = memo(
  ({
    width = 6,
    height = 22,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon, text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="screwdriver-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 6 22" fill="none">
          <Path
            d="M3 13V5M1 13h4v6a2 2 0 11-4 0v-6zM1 1h4L4 5H2L1 1z"
            stroke={color ?? isActive ? text.light : icon.tertiary}
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
