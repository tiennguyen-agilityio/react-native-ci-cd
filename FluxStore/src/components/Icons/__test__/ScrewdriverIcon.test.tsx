import {render, fireEvent, screen} from '@testing-library/react-native';

import {ScrewdriverIcon} from '../ScrewdriverIcon';

describe('ScrewdriverIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<ScrewdriverIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<ScrewdriverIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<ScrewdriverIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('screwdriver-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<ScrewdriverIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('screwdriver-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
