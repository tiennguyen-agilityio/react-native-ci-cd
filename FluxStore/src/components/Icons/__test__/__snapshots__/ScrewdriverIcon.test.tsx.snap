// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrewdriverIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="screwdriver-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={22}
    bbWidth={6}
    fill="none"
    focusable={false}
    height={22}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 22,
          "width": 6,
        },
      ]
    }
    vbHeight={22}
    vbWidth={6}
    width={6}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M3 13V5M1 13h4v6a2 2 0 11-4 0v-6zM1 1h4L4 5H2L1 1z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4288519581,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`ScrewdriverIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="screwdriver-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={22}
    bbWidth={6}
    fill="none"
    focusable={false}
    height={22}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 22,
          "width": 6,
        },
      ]
    }
    vbHeight={22}
    vbWidth={6}
    width={6}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M3 13V5M1 13h4v6a2 2 0 11-4 0v-6zM1 1h4L4 5H2L1 1z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
